JSONModel/JSONModel/JSONModel.m

```diff
@@ -692,6 +692,31 @@ -(void)__inspectProperties
             if (p && ![propertyIndex objectForKey:p.name]) {
                 [propertyIndex setValue:p forKey:p.name];
             }
+            
+            //generate custom setters
+            if (p) {
+                p.customSetters = [[NSMutableDictionary alloc] init];
+                for (Class allowedType in allowedJSONTypes) {
+                    NSString *className = NSStringFromClass([JSONValueTransformer classByResolvingClusterClasses:allowedType]);
+                    
+                    if (!p.customSetters[className]) {
+                        //check for a custom property setter method
+                        NSString* ucfirstName = [p.name stringByReplacingCharactersInRange:NSMakeRange(0,1)
+                                                                                withString:[[p.name substringToIndex:1] uppercaseString]];
+                        NSString* selectorName = [NSString stringWithFormat:@"set%@With%@:", ucfirstName, className];
+                        
+                        SEL customPropertySetter = NSSelectorFromString(selectorName);
+                        
+                        //check if there's a custom selector like this
+                        if ([self respondsToSelector: customPropertySetter]) {
+                            //cache the custom setter selector
+                            p.customSetters[className] = selectorName;
+                        } else {
+                            p.customSetters[className] = [NSNull null];
```

Comment:
No need for NSNull values as this code is run once - no handling needed for cached values